package.json

```diff
@@ -1,14 +1,17 @@
 {
+  "scripts": {
+    "build": "pnpm -r build",
+    "lint": "pnpm -r lint"
+  },
   "engines": {
     "node": "^18.20.2",
-    "npm": "^10.7.0"
+    "pnpm": "^9.14.3"
   },
   "devDependencies": {
     "prettier": "^3.1.1",
-    "prettier-plugin-solidity": "^1.3.1",
-    "typescript": "^5.5.4",
-    "@typescript-eslint/eslint-plugin": "^6.21.0",
-    "@typescript-eslint/parser": "^6.21.0",
-    "eslint": "^8.57.0"
-  }
+    "eslint": "^8.57.1",
+    "@typescript-eslint/eslint-plugin": "^8.16.0",
+    "@typescript-eslint/parser": "^8.16.0"
```

Comment:
I generally recommend managing tooling like prettier and eslint on each package and not at the root; this way when you have some package that has some dependency that prevents you from upgrading you can hold that package back without affecting the whole repo.

Pnpm catalogs are super useful for this kind of thing.

Reply:
I just put this here to make sure every internal package in the per repo, follows same eslint rules. I agree that later it can become harder to have same eslint rules for all packages.

Reply:
makes sense -- if you decide to eject what I usually do is have a common eslint config so each package can just reference that config, and use catalogs to manage the eslint & config dependencies.  Then you achieve all packages having shared eslint config, but can still easily exit out for specific places when you have a good reason to do so.

I follow this same pattern for all tools I use fwiw, including prettier, typescript, eslint, jest, etc.